diff --git a/sugar/activity/bundlebuilder.py b/sugar/activity/bundlebuilder.py
index 4ea05f8..3e70b3c 100644
--- a/sugar/activity/bundlebuilder.py
+++ b/sugar/activity/bundlebuilder.py
@@ -78,10 +78,10 @@ def _extract_bundle(source_file, dest_dir):
         if not os.path.exists(os.path.dirname(path)):
             os.makedirs(os.path.dirname(path))
 
-            outfile = open(path, 'wb')
-            outfile.write(zf.read(name))
-            outfile.flush()
-            outfile.close()
+        outfile = open(path, 'wb')
+        outfile.write(zf.read(name))
+        outfile.flush()
+        outfile.close()
 
 def _get_source_path(path=None):
     if path:
