diff --git a/src/sugar/activity/bundlebuilder.py b/src/sugar/activity/bundlebuilder.py
--- a/src/sugar/activity/bundlebuilder.py
+++ b/src/sugar/activity/bundlebuilder.py
@@ -35,7 +35,7 @@ from sugar import env
 from sugar.bundle.activitybundle import ActivityBundle
 
 
-IGNORE_DIRS = ['dist', '.git']
+IGNORE_DIRS = ['dist', '.git', 'debian', '.pc']
 IGNORE_FILES = ['.gitignore', 'MANIFEST', '*.pyc', '*~', '*.bak', 'pseudo.po']
 
 
