diff --git a/configure b/configure
index aac8564..c877ffd 100755
--- a/configure
+++ b/configure
@@ -19896,7 +19896,7 @@ PYGTK_DEFSDIR=`$PKG_CONFIG --variable=defsdir pygtk-2.0`
 
 # Setup GETTEXT
 #
-ALL_LINGUAS="af am ar ay bg bn bn_IN ca de dz el en es fa fa_AF ff fr gu ha hi ht ig is it ja km ko mk ml mn mr mvo ne nl pa pap pis pl ps pt pt_BR qu ro ru rw sd si te th tpi tr ur vi yo zh_CN zh_TW"
+ALL_LINGUAS="af am ar ay bg bn bn_IN ca da de dz el en es fa fa_AF ff fr gu ha hi ht ig is it ja km ko mk ml mn mr mvo ne nl pa pap pis pl ps pt pt_BR qu ro ru rw sd si te th tpi tr ur vi yo zh_CN zh_TW"
 GETTEXT_PACKAGE=sugar-toolkit
 
 case "$am__api_version" in
